spinalcordtoolbox/scripts/sct_dmri_denoise_patch2self.py

```diff
@@ -88,9 +89,13 @@ def main(argv: Sequence[str]):
         patch_radius = int(arguments.radius)
 
     file_to_denoise = arguments.i
-    output_file_name = arguments.o
-
-    path, file, ext = extract_fname(file_to_denoise)
+    if arguments.o is not None:
+        output_file_name_denoised = arguments.o
+        output_file_name_diff = add_suffix(arguments.o, "_difference")
+    else:
+        _, filename = os.path.split(file_to_denoise)
+        output_file_name_denoised = add_suffix(filename, "_patch2self_denoised")
+        output_file_name_diff = add_suffix(filename, "_patch2self_difference")
```

Comment:
Little code reorganizations like 6d835f18d9bff13e2d050166cc384c6df6988e09 are _so_ satisfying to me. :relaxed: :heart: